File: src/main/java/top/zbeboy/isy/web/vo/graduate/design/subject/GraduationDesignDeclareUpdateVo.java
package top.zbeboy.isy.web.vo.graduate.design.subject;

import lombok.Data;

import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

/**
 * Created by zbeboy on 2017/6/14.
 */
@Data
public class GraduationDesignDeclareUpdateVo {
    private Integer subjectTypeId;
    private Integer originTypeId;
    private Byte isNewSubject;
    private Byte isNewTeacherMake;
    private Byte isNewSubjectMake;
    private Byte isOldSubjectChange;
    private Integer oldSubjectUsesTimes;
    private String planPeriod;
    private String assistantTeacher;
    private String assistantTeacherAcademic;
    private String assistantTeacherNumber;
    private Integer guideTimes;
    private Integer guidePeoples;
    private Byte isOkApply;
    @NotNull
    private String graduationDesignPresubjectId;
    @NotNull
    @Size(max = 64)
    private String graduationDesignReleaseId;
    @NotNull
    private Integer staffId;
}

File: src/main/java/top/zbeboy/isy/web/vo/graduate/design/subject/GraduationDesignDeclareUpdateVo.kt
package top.zbeboy.isy.web.vo.graduate.design.subject

import javax.validation.constraints.NotNull
import javax.validation.constraints.Size

/**
 * Created by zbeboy 2018-01-26 .
 **/
open class GraduationDesignDeclareUpdateVo {
    var subjectTypeId: Int? = null
    var originTypeId: Int? = null
    var isNewSubject: Byte? = null
    var isNewTeacherMake: Byte? = null
    var isNewSubjectMake: Byte? = null
    var isOldSubjectChange: Byte? = null
    var oldSubjectUsesTimes: Int? = null
    var planPeriod: String? = null
    var assistantTeacher: String? = null
    var assistantTeacherAcademic: String? = null
    var assistantTeacherNumber: String? = null
    var guideTimes: Int? = null
    var guidePeoples: Int? = null
    var isOkApply: Byte? = null
    @NotNull
    var graduationDesignPresubjectId: String? = null
    @NotNull
    @Size(max = 64)
    var graduationDesignReleaseId: String? = null
    @NotNull
    var staffId: Int? = null
}